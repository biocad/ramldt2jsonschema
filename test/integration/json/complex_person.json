{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "Complex Person Schema",
  "type": "object",
  "definitions": {
    "address": {
      "type": "object",
      "properties": {
        "street_address": {"type": "string"},
        "city":           {"type": "string"},
        "state":          {"type": "string"},
        "planet":         {"type": ["string","null"]}
      },
      "required": ["street_address", "city", "state"]
    }
  },
  "properties": {
    "firstName": {
      "type": "string"
    },
    "lastName": {
      "type": "string"
    },
    "age": {
      "description": "Age in years",
      "type": "integer",
      "minimum": 0
    },
    "homeAddress": {"$ref": "#/definitions/address"},
    "workAddress": {"$ref": "#/definitions/address"},
    "relativesAddresses": {
      "type": "array",
      "items": {"$ref": "#/definitions/address"}
    },
    "favoriteMovie": {"type": "null"}
  },
  "required": ["firstName", "lastName", "homeAddress"]
}
