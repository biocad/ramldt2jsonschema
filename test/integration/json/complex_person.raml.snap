#%RAML 1.0 Library
types:
  Address:
    type: object
    properties:
      street_address: string
      city: string
      state: string
      planet?: string | nil
  TestType:
    type: object
    properties:
      firstName: string
      lastName: string
      age?:
        type: integer
        description: Age in years
      homeAddress: Address
      workAddress?: Address
      relativesAddresses?:
        type: array
        items:
          required: true
          type: Address
      favoriteMovie?: nil
    displayName: Complex Person Schema
